ogre2/src/media/materials/programs/heat_signature_fs.glsl

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#version 330
+
+// The input texture, which is set up by the Ogre Compositor infrastructure.
+uniform sampler2D RT;
+
+// input params from vertex shader
+in block
+{
+  vec2 uv0;
+} inPs;
+
+// final output color
+out vec4 fragColor;
+
+// The minimum and maximum temprature values (in Kelvin) that the
+// heat signature texture should be normalized to
+// (users can override these defaults)
+uniform float minTemp = 0.0;
+uniform float maxTemp = 100.0;
+
+// map a temperature from the [0, 655.35] range to the [minTemp, maxTemp]
+// range (655.35 is the largest Kelvin value allowed)
+float mapNormalized(float num)
+{
+  float mappedKelvin = ((maxTemp - minTemp) * num) + minTemp;
```

Comment:
Do you wan to clamp `num` to a maximum of `655.35` in case a large value is passed in? Also, is there a minium allowed value, such as 0 Kelvin?

Reply:
`num` is the result of the texture applied to the model, so that will always be a value between 0 and 1. `minTemp` and `maxTemp` should be checked though to make sure that they aren't outside of the [0, 655.35] range (these are specified by the user). I went ahead and added a check for this in https://github.com/ignitionrobotics/ign-rendering/pull/189/commits/d94e0dd9b7789d931a5c42bdf97f5e02d1c5c93e: https://github.com/ignitionrobotics/ign-rendering/blob/d94e0dd9b7789d931a5c42bdf97f5e02d1c5c93e/ogre2/src/Ogre2ThermalCamera.cc#L340-L355